conjur/util/credentials_from_file.py

```diff
@@ -89,19 +90,28 @@ def update_api_key_entry(self, user_to_update, credential_data, new_api_key,
         hosts[credential_data['machine']] = (user_to_update, None, new_api_key)
         self.build_netrc(netrc_obj)
 
-    def remove_credentials(self, conjurrc_conjur_url):
+    def remove_credentials(self, conjurrc):
         """
         Method that removes the described login entry from netrc
         """
-        netrc_data = self.load(conjurrc_conjur_url)
+        logging.debug(f"Attempting to remove credentials from '{DEFAULT_NETRC_FILE}'")
+        # pylint: disable=no-else-return
+        if not os.path.exists(DEFAULT_NETRC_FILE):
+            return
+        elif os.path.exists(DEFAULT_NETRC_FILE) and os.path.getsize(DEFAULT_NETRC_FILE) != 0:
+            netrc_data = self.load(conjurrc.conjur_url)
+
+            netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
+            hosts = netrc_obj.hosts
+            hosts.pop(netrc_data['machine'], None)
 
-        netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
-        hosts = netrc_obj.hosts
-        hosts.pop(netrc_data['machine'], None)
+            netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
+            netrc_obj.hosts.pop(netrc_data['machine'], None)
+            self.build_netrc(netrc_obj)
```

Comment:
not follow why do we need 
` 
netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)` 
           `  hosts = netrc_obj.hosts` 
          `  hosts.pop(netrc_data['machine'], None)`

Reply:
this happens on logout when they user wants to remove their credentials from the netrc and 'logout'. We basically find the machine and 'pop' it and rebuild the netrc here without the group the user wants to remove